### PR TITLE
[tests-only] E2e test public link

### PR DESCRIPTION
**Pull request by SwikritiT**
## Related Issue

- part of: https://github.com/owncloud/web/issues/6389

## How Has This Been Tested?

- Locally
- CI

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [ ] ...

### CODE REVIEW

**Comment by ownclouders** [comment]
Results for <strong>e2e-tests oC10</strong> https://drone.owncloud.com/owncloud/web/25372/11/1
<details><summary>:boom: To see the trace, please open the link in the console ...</summary>

<p>


#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/25372/tracing/alice-can-share-this-weeks-meal-plan-with-all-parents-alice-2022-5-6-07-37-23.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/25372/tracing/alice-can-share-this-weeks-meal-plan-with-all-parents-brian-2022-5-6-07-37-41.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/25372/tracing/alice-can-share-this-weeks-meal-plan-with-all-parents-carol-2022-5-6-07-37-46.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/25372/tracing/folder-alice-2022-5-6-07-35-19.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/25372/tracing/folder-brian-2022-5-6-07-35-26.zip 


</p></details>

File: tests/e2e/support/objects/app-files/resource/actions.ts [hunk 1]
```diff
@@ -112,10 +111,9 @@ export const downloadResources = async (args: downloadResourcesArgs): Promise<Do
 
   switch (via) {
     case 'SIDEBAR_PANEL': {
-      if (folder) {
+      if (folder && folder !== 'undefined') {
```
Comment (saw-jan) [inline tests/e2e/support/objects/app-files/resource/actions.ts:114]: are you sure that folder will have string `undefined`?

File: tests/e2e/support/objects/app-files/resource/actions.ts [hunk 2]
```diff
@@ -76,8 +76,7 @@ export interface uploadResourceArgs {
 
 export const uploadResource = async (args: uploadResourceArgs): Promise<void> => {
   const { page, resources, to, createVersion } = args
-
-  if (to) {
+  if (to && to !== 'undefined') {
```
Comment (saw-jan) [inline tests/e2e/support/objects/app-files/resource/actions.ts:79]: are you sure that folder will have string undefined?

File: tests/e2e/support/objects/app-files/resource/actions.ts (hunk 1 above)
Comment (SwikritiT) [reply tests/e2e/support/objects/app-files/resource/actions.ts:114]: yes if there's no `from` column in data table
https://github.com/owncloud/web/blob/841d6af28b2b34f23138ce8c6b0244098ce4a978/tests/e2e/cucumber/features/integrations/link.feature#L38-L41

File: tests/e2e/cucumber/steps/app-files/page/public.ts [hunk 3]
```diff
@@ -63,6 +66,63 @@ When(
...
+      await pageObject.upload({
+        to: info.to,
+        resources: [this.filesEnvironment.getFile({ name: info.resource })],
+        createVersion: info.create_version === 'true'
+      })
+    }
+  }
+)
+
+When(
```
Comment (kiranparajuli589) [inline tests/e2e/cucumber/steps/app-files/page/public.ts:123]: is this a then step?

File: tests/e2e/cucumber/steps/app-files/resource.ts [hunk 4]
```diff
@@ -170,3 +120,60 @@ When(
...
   }
 )
+
+export const processDownload = async (
+  stepTable: DataTable,
+  pageObject: any,
+  actionType: string
+) => {
+  let downloads
+  let files, parentFolder
```
Comment (kiranparajuli589) [inline tests/e2e/cucumber/steps/app-files/resource.ts:130]:
```suggestion
  let downloads, files, parentFolder
```

File: tests/e2e/cucumber/steps/app-files/resource.ts [hunk 5]
```diff
@@ -170,3 +120,60 @@ When(
...
+
+    return acc
+  }, {})
+
+  for (const folder of Object.keys(downloadInfo)) {
+    files = downloadInfo[folder]
+    parentFolder = folder
+    downloads = await pageObject.download({
+      folder,
+      names: files,
```
Comment (saw-jan) [inline tests/e2e/cucumber/steps/app-files/resource.ts:148]: > yes if there's no `from` column in data table

I would suggest you to do that here.
```suggestion
    parentFolder = folder !== 'undefined' ? folder : null
    downloads = await pageObject.download({
      folder: parentFolder,
      names: files,
```

File: tests/e2e/cucumber/features/integrations/link.feature [hunk 6]
```diff
@@ -27,13 +27,27 @@ Feature: link
...
     #Then the public should not see the following files on the files-drop page
     #  | textfile.txt |
-    And "Anonymous" logs out
     When "Alice" downloads the following resources using the batch action
       | resource     | from         |
       | lorem.txt    | folderPublic |
       | textfile.txt | folderPublic |
     And "Alice" edits the public link named "myPublicLink" of resource "folderPublic" changing role to "editor"
+    And "Anonymous" refreshes the old link
+    And "Anonymous" downloads the following public link resources using the sidebar panel
```
Comment (saw-jan) [inline tests/e2e/cucumber/features/integrations/link.feature:38]: check whether `batch action` works or not.

File: tests/e2e/support/objects/app-files/resource/actions.ts (hunk 1 above)
Comment (saw-jan) [reply tests/e2e/support/objects/app-files/resource/actions.ts:114]: @SwikritiT see https://github.com/owncloud/web/pull/6889#discussion_r871032039

File: tests/e2e/cucumber/features/integrations/link.feature (hunk 6 above)
Comment (ScharfViktor) [reply tests/e2e/cucumber/features/integrations/link.feature:38]: Just wondering how do you suggest to check that batch action works?  It checked here:`expect(download.suggestedFilename()).toBe('download.tar')`.

Comment (SwikritiT) [reply tests/e2e/cucumber/features/integrations/link.feature:38]: @saw-jan as Viktor mentioned above we are asserting that the `zip` or `tar` file is downloaded in the when step itself.

Comment (saw-jan) [reply tests/e2e/cucumber/features/integrations/link.feature:38]: Yeah, saw that one :+1:
Just wanted to make sure that batch download in the public link also works but not to introduce the test step itself.

Comment (SwikritiT) [reply tests/e2e/cucumber/features/integrations/link.feature:38]: The batch action for the public link doesn't work https://github.com/owncloud/web/issues/6423

**Review by saw-jan — COMMENTED** [review]
LGTM

**Review by ScharfViktor — APPROVED** [review]

**Pull request merged**